GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1693,12 +1693,12 @@ public struct DrawRefArgs
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-                        var gravatar = _avatarCache.GetImage($"{revision.AuthorEmail}.png", null);
+                        var gravatar = _gravatarService.GetImage(revision.AuthorEmail, null);
```

Comment:
My perception of this class was more generic, but it did cross my mind few times whether this generic implementation is really warranted given there are no other image cache requirements.
I agree - it would simplify things, the name will need to be changed to reflect the functionality. Something like `AvatarDirectoryCache` which will be operating on emails instead of traditional filenames. 

In mean I have pushed an update, please have a look. Everything seems to be working nicely on Windows. I will try it on a linux when time permits.